### PR TITLE
Refactoring Check slightly to clean it up

### PR DESCRIPTION
**Pull request by Barthelemy**
- Introduce CheckConfig to encapsulate the configuration of a check
- Avoid code dupcliation for module loading and class instantiating

This is in view of https://alice.its.cern.ch/jira/browse/QC-299

### CODE REVIEW

File: Framework/src/Check.cxx [hunk 1]
```diff
@@ -307,8 +264,9 @@ std::shared_ptr<QualityObject> Check::check(std::map<std::string, std::shared_pt
       mLatestQuality->updateQuality(mCheckInterface->check(&shadowMap));
     }
   }
-  mLogger << mName << " Quality: " << mLatestQuality->getQuality() << AliceO2::InfoLogger::InfoLogger::endm;
+  mLogger << mCheckConfig.checkName << " Quality: " << mLatestQuality->getQuality() << AliceO2::InfoLogger::InfoLogger::endm;
   // Trigger beautification
+  cout << __FILE__ << ":" << __LINE__ << " - calling beautify" << endl;
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/src/Check.cxx]: yes, left over. Fixed

File: Framework/src/Check.cxx [hunk 2]
```diff
@@ -210,60 +212,15 @@ void Check::initPolicy(std::string policyType)
...
+  mCheckInterface->configure(mCheckConfig.checkName);
 
   /** 
    * The policy needs to be here. If running in constructor, the lambda gets wrong reference
    * and runs into SegmentationFault.
    */
-  initPolicy(mPolicyType);
-}
-
-void Check::loadLibrary()
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/src/Check.cxx]: done

**Comment by Barthelemy** [comment]
Sorry for the crash I reverted too many things at some point before pushing. It should work now

**Comment by knopers8** [comment]
Confirmed, it seems to work now.

**Comment by knopers8** [comment]
Missclick

**Pull request merged**